text/0000-module-unification.md

```diff
@@ -0,0 +1,837 @@
+should be considered to share a common namespace. Routes, controllers, and
+templates are a good example, as are models, adapters, and serializers.
+
+A related concept to understand about collections is the notion of a default
+type. Every top-level module within a collection can be considered to match its
+default type (unless named exports are used in those modules to represent types
+other than the default). Within a collection's namespaces, every module must be
+either that default type or related to it. It's helpful to consider that every
+namespace within a collection represents a set of named module exports, and that
+the default type represents the default export for that collection.
+
+Here's an illustration of exports from a collection:
+
+```
+src
+  data
+    models
+      author.js <- exports an Author `model`, the default type in the `models` collection
+      comment
+        adapter.js     <- exports a Comment `adapter`
+        model.js       <- exports a Comment `model`
+        serializer.js  <- exports a Comment `serializer`
+```
+
+#### Components
+
+The term "component" has been widely adopted across most front-end frameworks
+to describe a broad swath of UI concerns. Using the same term for the collection
+of template-invocable UI elements will lower the learning curve for developers
+who are new to Ember, while allowing for a useful set of specialized terms to
+flourish to describe particular _types_ of components.
+
+We've already started down the road of component specialization by introducing
+the concept of "routable components". Once we start actually using "routable
+components" in practice, it will become necessary to refer to plain old
+components as something more specific, like "template components". And this
+distinction will probably lead to plain old helpers being referred to as
+"template helpers". Other concepts, such as "Glimmer components" and "template
+component modifiers" will soon be mixed in. We will end up with a multi-faceted
+toolbox available at the template layer which deserves a simple name that
+matches developer expectations. The general term "components" seems a good fit.
+
+### Scope
+
+Developers should understand the available levels of module scope, as well as
+when each is appropriate to use. Scope should be considered when modules are
+generated, and developers should feel free to move modules if they expand or
+contract in scope.
+
+The following levels of scope should be understood:
+
+* Private - private collections should be used when a component or utility
+  function is needed from a single namespace.
+
+* Project - top-level, project-wide collections should be used for modules that
+  are needed throughout a project.
+
+* Local package - namespaced collections can be useful to group a common set of
+  cross-cutting concerns within a project.
+
+* Local engine - a type of local package that encapsulates a set of
```

Comment:
> Or does the (new) Addon type export eliminate that from concern?

Right. The intent is to make the distinction between apps, engines, and addons explicit via the main export.